src/App.tsx

```diff
@@ -2,13 +2,25 @@ import React from 'react';
 
 import './App.scss';
 
-// import posts from './api/posts';
-// import comments from './api/comments';
-// import users from './api/users';
+import posts from './api/posts';
+import comments from './api/comments';
+import users from './api/users';
+import { PostList } from './components/PostList';
+
+const preparedPosts = posts.map(post => {
+  const modifiedPost = {
+    ...post,
+    user: users.find(user => user.id === post.userId) || null,
+    comments: comments.filter(comment => comment.postId === post.id),
```

Comment:
So, your comments field can't be null, cause filter returns empty array if no elements had passed the check